lib/src/reducers/useSchemaReducer.tsx

```diff
@@ -0,0 +1,161 @@
+import { ObjectFieldTemplateProps } from '@rjsf/core';
+import React, { useCallback, useReducer } from 'react';
+import { VernaSchemaType } from ':/types/rjsf';
+import { TranslationType } from ':/types/translations';
+import { Maybe } from ':/types/utils';
+import { ReducerAction, VernaActionsEnum } from ':/types/VernaProvider';
+import ShowCaseWidgetProps from ':/types/Widgets';
+import { cleanUiSchema, reduceSchema } from ':/utils/schema';
+
+function schemaReducer(state: VernaSchemaType, action: ReducerAction): VernaSchemaType {
+  if (action.error) {
+    throw new Error('An error happened while applying a modification on the schema.');
+  } else {
+    const selector: Maybe<string> = action.meta?.selector;
+
+    if (selector) {
+      const reducedSchema: Maybe<VernaSchemaType> = reduceSchema(
+        {
+          schema: state.schema?.properties?.[selector],
+          translationSchema: state.translationSchema,
+          uiSchema: state.uiSchema?.[selector],
+        },
+        action,
+      );
+      if (!reducedSchema) throw new Error('Error, something went wrong while modifying the schema');
+
+      return {
+        schema: {
+          ...state.schema,
+          properties: { ...state.schema?.properties, [selector]: reducedSchema.schema || {} },
+        },
```

Comment:
I don't think it will works, cause if i delete a property of a widget, when the reducer goes on it will merge the new state without the property inside of the old state that still have the property, on my version i ignore the old selected section and place only the new one.